### DISCUSSION TITLE
MACRO tooltip

### DISCUSSION THREAD

**Discussion by dts-1990** [eclipse-cdt/cdt#465] [Ideas]
If the code in side macro is really long, it is hard to find what is the condition of #endif. For example:

```
#ifdef MACRO_NAME
// Some code
#endif 
```
We have to scroll up until #ifdef line to see. It is even more harder if we have nested macros.

Can we add a feature that when we hover the mouse on #endif it will show a tooltip containing the info of "#ifdef MACRO_NAME"?

**Comment by Kugman** [comment]
in visual studio its already exists.
when your mouse point on the #endif, you can see the condition in a small tooltip.
are you working with VS code?

**Comment by gvl2023** [comment]
A keyboard shortcut to jump between ifdef/endif would also be nice, there is a long outstanding feature request for this, 18 years to be precise. I'm still waiting for it to be implemented. https://bugs.eclipse.org/bugs/show_bug.cgi?id=182579
